code/WorkInProgress/AzrunStuff.dm

```diff
@@ -239,3 +239,353 @@
+			target_client = statusargs["client"]
+
+	onUpdate()
+		..()
+		get_back_here()
+
+	move_trigger(mob/user, ev)
+		. = 0
+		get_back_here()
+
+	proc/get_back_here()
+		var/mob/dead/observer/ghost = owner
+		if(istype(ghost) && bound_target && ghost.loc != bound_target)
+			boutput(ghost, "You find yourself pulled back into [bound_target]!")
+			ghost.set_loc(bound_target)
+
+	onRemove()
+		..()
+		var/mob/dead/observer/ghost = owner
+		if(istype(ghost) && bound_target && ghost.loc == bound_target)
+			ON_COOLDOWN(bound_target, "ghost_suck", 2 SECONDS)
+			ghost.set_loc(get_turf(bound_target))
+
+
+/obj/item/organ/brain/ghost/afterattack(atom/target, mob/user)
+	if(istype(target, /obj/machinery/bot))
+		target.AddComponent(/datum/component/brain_control, src, user)
+
+/obj/item/organ/brain/ghost/mouse_drop(atom/over_object, src_location, over_location, over_control, params)
+	if(istype(over_object, /obj/machinery/bot))
+		afterattack(over_object, usr)
+
+/datum/component/brain_control
+	var/orig_path
+	var/obj/item/organ/brain/controller
+
+TYPEINFO(/datum/component/brain_control)
+	initialization_args = list()
+
+TYPEINFO(/datum/component/controlled_by_mob)
+	initialization_args = list(
+		ARG_INFO("B", DATA_INPUT_REFPICKER, "Brain to enter"),
+		ARG_INFO("user", DATA_INPUT_MOB_REFERENCE, "Mob to control the component")
+	)
+
+/datum/component/brain_control/Initialize(obj/item/organ/brain/B, mob/user)
+	var/atom/target = parent
+	if(!istype(target))
+		return COMPONENT_INCOMPATIBLE
+
+	orig_path = parent.type
+	if(istype(B))
+		controller = B
+	else
+		return COMPONENT_INCOMPATIBLE
+
+	// /obj/machinery/bot/medbot  TODO?
+	// /obj/machinery/bot/cleanbot <-> /mob/living/critter/bot/cleanbot
+	// /obj/machinery/bot/firebot <-> /mob/living/critter/bot/firebot
+	// /obj/machinery/bot/floorbot  TODO?
+	var/obj/machinery/bot/new_bot
+	if(istype(target, /obj/machinery/bot)) // Maybe move to /obj/machinery/bot to get mapping out of here
+		if(istype(target, /obj/machinery/bot/cleanbot ))
+			new_bot = new /mob/living/critter/bot/cleanbot(target.loc)
+		else if(istype(target, /obj/machinery/bot/firebot ))
+			new_bot = new /mob/living/critter/bot/firebot(target.loc)
+
+	if(new_bot)
+		qdel(target)
+		parent = new_bot
+		RegisterSignal(parent, list(COMSIG_ATOM_POST_UPDATE_ICON), .proc/update_icon)
+		RegisterSignal(parent, list(COMSIG_ATTACKBY), .proc/check_attack)
+
+		if (controller.owner) //Mind transfer also handles key transfer.
+			controller.owner.transfer_to(new_bot)
+		user.u_equip(controller)
+		controller.set_loc(new_bot)
+		new_bot.UpdateIcon()
+	else
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/brain_control/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_ATTACKBY, COMSIG_ATOM_POST_UPDATE_ICON))
+	. = ..()
+
+/datum/component/brain_control/proc/update_icon(atom/A)
+	var/image/I = A.SafeGetOverlayImage("brain",image('icons/obj/items/device.dmi', "head-brain"))
+	I.appearance_flags = RESET_COLOR | KEEP_APART
```

Comment:
```suggestion
	I.appearance_flags = RESET_COLOR | KEEP_APART | PIXEL_SCALE
```
PIXEL_SCALE should be pretty universally applied, in case of scaling makes things look better for pixel art.